community-connectors/google-fit/GoogleFit.js

```diff
@@ -0,0 +1,255 @@
+    "WALKING (FITNESS)":                          93,
+    NORDING_WALKING:                              94,
+    "WALKING (TREADMILL)":                        95,
+    WATERPOLO:                                    96,
+    WEIGHTLIFTING:                                97,
+    WHEELCHAIR:                                   98,
+    WINDSURFING:                                  99,
+    YOGA:                                         100,
+    ZUMBA:                                        101,
+
+    "OTHER (UNCLASSIFIED FITNESS ACTIVITY)":      108
+  };
+
+  // Create a reverse map of activity codes to name.
+  this.ACTIVITY_MAP = {};
+
+  for (var key in this.ACTIVITY_TYPE) {
+    if (this.ACTIVITY_TYPE.hasOwnProperty(key)) {
+      this.ACTIVITY_MAP[this.ACTIVITY_TYPE[key]] = key;
+    }
+  }
+}
+
+/**
+ * Gets the description for the activity given the activity ID.
+ * @param {number} The activity ID
+ * @return {string} The activity description
+ */
+GoogleFit.prototype.getActivityDescription = function(id) {
+  return this.ACTIVITY_MAP[id];
+};
+
+/**
+ * Gets an OAuth2 service abject for Google Fit.
+ *
+ * @return {Object} OAuth2 library service object.
+ */
+GoogleFit.prototype.getService = function() {
+  if (this._service) {
+    return this._service;
+  }
+  // Create a new service with the given name. The name will be used when
+  // persisting the authorized token, so ensure it is unique within the
+  // scope of the property store.
+  this._service = OAuth2.createService('fitness')
+
+      // Set the endpoint URLs, which are the same for all Google services.
+      .setAuthorizationBaseUrl('https://accounts.google.com/o/oauth2/auth')
+      .setTokenUrl('https://accounts.google.com/o/oauth2/token')
+
+      // Set the client ID and secret, from the Google Developers Console.
+      .setClientId(this.config.OAUTH_CLIENT_ID)
+      .setClientSecret(this.config.OAUTH_CLIENT_SECRET)
+
+      // Set the name of the callback function in the script referenced
+      // above that should be invoked to complete the OAuth flow.
+      .setCallbackFunction('authCallback')
+
+      // Set the property store where authorized tokens should be persisted.
+      .setPropertyStore(PropertiesService.getUserProperties())
+
+      // Set the scopes to request (space-separated for Google services).
+      .setScope('https://www.googleapis.com/auth/fitness.activity.read https://www.googleapis.com/auth/fitness.body.read')
+
+      // Below are Google-specific OAuth2 parameters.
+
+      // Sets the login hint, which will prevent the account chooser screen
+      // from being shown to users logged in with multiple accounts.
+      .setParam('login_hint', Session.getActiveUser().getEmail())
+
+      // Requests offline access.
+      .setParam('access_type', 'offline')
+
+      // Forces the approval prompt every time. This is useful for testing,
+      // but not desirable in a production application.
+      .setParam('approval_prompt', 'force');
```

Comment:
This param should be removed before publication.

Reply:
Done.